packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -931,9 +931,9 @@ export class ForkChoice implements IForkChoice {
    * Return only the non-finalized blocks.
    */
   getAllAncestorBlocks(blockRoot: RootHex): ProtoBlock[] {
-    const blocks = this.protoArray.getAllAncestorNodes(blockRoot);
-    // the last node is the previous finalized one, it's there to check onBlock finalized checkpoint only.
-    return blocks.slice(0, blocks.length - 1);
+    const ancestors = this.protoArray.getAllAncestorNodes(blockRoot);
+    // Exclude the last node as before
```

Comment:
the comment above `the last node is the previous finalized one, it's there to check onBlock finalized checkpoint only.`  is important, please keep it

this comment "Exclude the last node as before" is not meaningful to put in the implementation, suggest to put it as github PR's comment instead